### PR TITLE
use the global default socket path for spire-agent run

### PR DESCRIPTION
**Pull request by joewilliams**
**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
Default socket path for `spire-agent run`.

**Description of change**
Make the default spire-agent default socket path more obvious by using the global default value.

**Which issue this PR fixes**

### CODE REVIEW

**Comment by azdagron** [comment]
Code change is looking good!

There is a small risk of breaking backwards compatibility here for folks who don't define the `socket_path` in their agent configuration who were relying on the strange `./spire_api` default. Considering all of the configurations defined by tutorials and examples do define socket_path, it doesn't seem too likely that someone would be reliant on this, but I thought I'd call out the risk.

Thoughts @evan2645 , @amartinezfayo , @APTy , @mcpherrinm? Any concern taking this potentially breaking change? What is your evaluation of the risk?

**Comment by joewilliams** [comment]
Not sure why this test is failing while the others are passing:

```
golangci/golangci-lint info installed /home/travis/gopath/src/github.com/spiffe/spire/.build/linux-x86_64/golangci_lint/v1.27.0/golangci-lint

cmd/spire-agent/cli/run/run.go:23: File is not `goimports`-ed (goimports)

	"github.com/sirupsen/logrus"

Makefile:377: recipe for target 'lint-code' failed
```

**Comment by azdagron** [comment]
`goimports` orders the imports alphabetically, which would place `"github.com/spiffe/spire/cmd/spire-agent/cli/common"` just after `"github.com/sirupsen/logrus"`.

**Comment by amartinezfayo** [comment]
> Thoughts @evan2645 , @amartinezfayo , @APTy , @mcpherrinm? Any concern taking this potentially breaking change? What is your evaluation of the risk?

I'm OK with this change. I see the risk of breaking something really small and I think that it's good to have this fixed before 1.0.
I would make sure that we include this change in the CHANGELOG for the release so anyone updating is aware of the change.

File: doc/spire_agent.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ This may be useful for templating configuration files, for example across differ
 | `log_format`              | Format of logs, \<text\|json\>                                        | Text                 |
 | `server_address`          | DNS name or IP address of the SPIRE server                            |                      |
 | `server_port`             | Port number of the SPIRE server                                       |                      |
-| `socket_path`             | Location to bind the workload API socket                              | $PWD/spire_api       |
+| `socket_path`             | Location to bind the workload API socket                              | /tmp/agent.sock      |
```
Comment (amartinezfayo) [inline doc/spire_agent.md:48]: The default value described in agent_full.conf needs to be updated also.
https://github.com/spiffe/spire/blob/master/conf/agent/agent_full.conf#L31

Comment (joewilliams) [reply doc/spire_agent.md:48]: Got it, thanks!

**Comment by mcpherrinm** [comment]
While I think this is a technically a breaking change, I think it's worth fixing this to a better default now.

**Comment by azdagron** [comment]
@joewilliams , thanks for putting this together! We (maintainers) huddled and discussed this offline. I think we are ok accepting the small risk. If you wouldn't mind fixing up the sign-off on the commits (and maybe squash and rebase), we're ready to get this merged.

**Comment by joewilliams** [comment]
@azdagron okay, got things cleaned up.

**Pull request merged**